test/extended/util/framework.go

```diff
@@ -238,10 +238,56 @@ func processScanError(log string) error {
 	return fmt.Errorf(log)
 }
 
+// WaitForImageStreams creates custom ruby imageStream in OpenShift namespace
+// TODO: To eliminate the dependency on OpenShift Samples Operator in future,
+// WaitForImageStreams should be a replacement of WaitForOpenShiftNamespaceImageStreams func
+func WaitForImageStreams(oc *CLI) error {
+	ctx := context.Background()
+	var registryHostname string
+
+	images := map[string]string{
+		"ruby": "registry.access.redhat.com/ubi8/ruby-33:latest",
+	}
+
+	// Check to see if we have ImageRegistry enabled
+	hasImageRegistry, err := IsCapabilityEnabled(oc, configv1.ClusterVersionCapabilityImageRegistry)
+	if err != nil {
+		return err
+	}
+	if hasImageRegistry {
+		registryHostname, err = WaitForInternalRegistryHostname(oc)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Create custom imageStream using `oc import-image`
+	e2e.Logf("waiting for imagestreams to be imported")
+	for image, location := range images {
+		err := oc.Run("import-image").Args(image, fmt.Sprintf("--from=%s", location), "--confirm").Execute()
```

Comment:
please add --import-mode=PreserveOriginal
it'll cause issues on arm64

```

    --import-mode='Legacy':
        Imports the full manifest list of a tag when set to 'PreserveOriginal'. Defaults to
        'Legacy'.
```

Reply:
I do think your fix was resolved. You just hit a few flakes on gcp. Most of those are pretty common.

Reply:
Any intent to add `--import-mode=PreserveOriginal` back?